src/components/Table/components/cell.tsx

```diff
@@ -34,6 +34,8 @@ export type GridCellContent = {
   revealOnRowHover?: true;
   /** Tooltip to add to a cell */
   tooltip?: ReactNode;
+  /** Allows cell to be editable when hovering, and also highlights the field on hover */
+  editableOnHover?: true;
```

Comment:
**Nit/suggestion**: `editableOnHover` doesn't really tell me whats happening here. Really we're just putting a border around fields on hover. The field doesn't actually become editable on hover, it just has a border now 🤷. 
Taking a step back, this feels like something that **should be globally applied**, not just for one or two tables with inputs. I would request more design direction here. The issue with Blueprint is that every page is so different. This is just going to add to that problem.
Also, when a user uses their `tab` key to go between fields, then we don't get the same "bordered" treatment, unless they're hovering that row. As this PR stands, it's creating an inconsistent experience that I don't believe provides a better UX for our users. I would ask Design to fully flush this idea out, and figure out if this should be the global pattern. If not the global pattern, have rules that define when to use this pattern vs not. We should be doing this as it is best for our users.

Reply:
Ah damn, I was pushing towards "each table itself should at least be consistent" (instead of a per cell decision), but ngl I like Brandon's push about "do they just want all tables in BP to act this way?"

Because that is actually easier for us to globally roll out, and per Brandon's point better for users (consistency), to just have tables "always do that" (whatever the latest/greatest UX is), vs. picking tables onsey-twosy that do/do not get the special behavior.

Reply:
AFAIK this is a new design thing, I can ask design if the will be new standard, I tried not to impose massive changes, this is the figma that I've been following for this PR: https://www.figma.com/design/xQ3iNLHJCFEJeOi7boONY4/Q1-2024-%7C-Dynamic-Schedules?node-id=11225-33807&t=BgAyiY5hjDNYwF9v-1

Reply:
@0ces cool, feel free to loop me into any design discussions

Reply:
Yes, this is supposed to be the new global standard.

Reply:
What does this comment mean by "non-table cells"?